static/src/javascripts/projects/common/modules/commercial/article-body-adverts.js

```diff
@@ -62,19 +72,29 @@ define([
 
             if (config.page.hasInlineMerchandise) {
                 adNames.unshift(['im', 'im']);
-                insertAdAtP(spacefinder.getParaWithSpace(lenientRules));
-            }
-            insertAdAtP(spacefinder.getParaWithSpace(rules));
 
-            if (detect.isBreakpoint({ max: 'tablet' })) {
-                insertAdAtP(spacefinder.getParaWithSpace(rules));
+                inlineMercPromise = spacefinder.getParaWithSpace(lenientRules).then(function (space) {
+                    return insertAdAtP(space);
+                });
+            } else {
+                inlineMercPromise = Promise.resolve(null);
             }
+
+            return inlineMercPromise.then(function () {
```

Comment:
This is one of the main reasons we should use React btw. This code is horrid because we store loads of crap in the DOM and expect to be able to find it later. The state should all be in JavaScript data structures, not in the DOM.